sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ContainerThroughputControlGroupProperties.java

```diff
@@ -51,15 +53,33 @@ public int enableThroughputControlGroup(ThroughputControlGroupInternal group) {
             this.supressInitErrorGroupSet.add(group.getGroupName());
         }
 
+        final AtomicBoolean updatedGroupConfig = new AtomicBoolean(false);
+
         // Only throw when two different groups are using the same id (databaseId + containerId + groupName)
-        if (this.throughputControlGroupSet.stream()
-                .anyMatch(existingGroup -> Objects.equals(existingGroup.getId(), group.getId()) && !existingGroup.equals(group))) {
-            throw new IllegalArgumentException("Throughput control group with id " + group.getId() + " already exists");
-        }
+        this.throughputControlGroupSet.stream()
+            .forEach(existingGroup -> {
+                if (Objects.equals(existingGroup.getId(), group.getId()) &&
+                    !existingGroup.equals(group)) {
+
+                    if (existingGroup.isDefault() != group.isDefault() ||
+                        existingGroup.isContinueOnInitError() != group.isContinueOnInitError()) {
+                        throw new IllegalArgumentException(
+                            "Throughput control group with id " + group.getId() + " already exists with different " +
+                            "setting for isDefault or isContinueOnInitError - these properties are immutable and " +
+                            "cannot be changed.");
+                    }
+
+                    this.throughputControlGroupSet.remove(existingGroup);
+                    logger.info(
+                        "Throughput control group with id {} already exists with different config. Will update config.",
+                        group.getId());
+                    updatedGroupConfig.set(true);
+                }
+            });
 
         this.throughputControlGroupSet.add(group);
```

Comment:
hmm, can we change to return true if we ended up adding the group into the groups? 
``` 
     if (this.throughputControlGroupSet.add(group)) {
        updatedGroupConfig.set(true)     
    }
``` 

And then later,  
``` 
            if ((groupSizeAfter > groupSizeBefore && groupSizeAfter == 1) || wasGroupConfigUpdated) {
``` 

can be changed into 
``` 
            if (groupSizeAfter == 1 && wasGroupConfigUpdated) {
```